test/frontend_test.dart

```diff
@@ -0,0 +1,457 @@
+            '/a/span/font/font[contains(text(),"egress")]'));
+        expect(futureElementExists(logoutItem), true);
+      });
+    });
+
+    group("probaidoszak page tests |", () {
+      setUpAll(() async {
+        await waitForAngular(driver);
+        await loginUserWithCredentials(
+            newUserEmail, allUserPassword, driver, true);
+        await navigateToPage(driver, 'profile/product/probaidoszak');
+        await waitForAngular(driver);
+        await sleep5();
+      });
+
+      test('you see a bunch of videos showing up', () async {
+        var videoItemsLength = await driver
+            .findElements(const By.cssSelector(
+                'video-summary-list > div.training-type-list > div '
+                '> video-summary > div.video-summary'))
+            .length;
+        print("Video items found: " + videoItemsLength.toString());
+        expect(await isXAtLeastValueOfY(videoItemsLength, 5), true);
+      });
+
+      test('video popup shows on page after clicking a video', () async {
+        var firstVideoItem;
+        firstVideoItem = await driver.findElement(const By.xpath(
+            '*//video-summary-list/div[contains(@class, '
+                '"training-type-list")]//video-summary[1]'));
+        await firstVideoItem.click();
+        await sleep5();
+        var videoPlayer = await driver.findElement(const By.xpath(
+            '//*[@id="default-acx-overlay-container"]//vimeo-player/'
+            'div[contains(@class,"modal-body")]/div/div/iframe'));
+        await videoPlayer.click();
+        await sleep2();
+        await waitForAngular(driver);
+        expect(await elementExists(videoPlayer), true);
+      });
+
+      group("probaidoszak filter tests |", () {
+        setUp(() async {
+          await navigateToPage(driver, 'profile/product/probaidoszak');
+          await waitForAngular(driver);
+          await sleep5();
+        });
+
+        test('joga filter options work', () async {
+          var workoutVarietyFilterDropdown = await driver.findElement(
+              const By.xpath('*//tag-set-filter/material-dropdown-select/'
+                  'dropdown-button/div/span[contains(text(), "Edz")]'));
+          await workoutVarietyFilterDropdown.click();
+          var jogaFilterOption = await driver.findElement(const By.xpath(
+              '*//div[contains(@class, "options")]/div/material-select-'
+                  'dropdown-item/span[contains(text(), "jóga")]'));
+          await jogaFilterOption.click();
+          await waitForAngular(driver);
+          await sleep2();
+          var filteredVideoItemsLength = await driver
+              .findElements(const By.cssSelector(
+                  'video-summary-list > div.training-type-list > div '
+                  '> video-summary > div.video-summary'))
+              .length;
+          expect(await isXAtLeastValueOfY(filteredVideoItemsLength, 2), true);
```

Comment:
This is not really a good test condition. This will probably be true even if the filter does not work. How about checking the workout class ("Edzésfajta") inner text content in the filtered video summary?